### PR TITLE
Added spawn eggs.

### PR DESCRIPTION
**Pull request by spacechase0**
Signed-off-by: Chase W chasew@silkweaver.com

### CODE REVIEW

File: src/main/java/org/spout/vanilla/material/item/SpawnEgg.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+public class SpawnEgg extends GenericItem {
+
+	public SpawnEgg(String name, int id, int data) {
+		super(name, id, data);
+	}
+	
+	@Override
+	public void onInteract(Entity entity, Point position, Action type, BlockFace clickedFace) {
+		Class<? extends VanillaEntity> controller = org.spout.vanilla.entity.Entity.getByID((int) getData()).getController();
```
Comment (zml2008) [inline src/main/java/org/spout/vanilla/material/item/SpawnEgg.java]: I'd move this into the constructor, since there is one instance of SpawnEgg per damage value. Also, you only updated the sheep spawn egg. Was that intentional?

Comment (spacechase0) [inline src/main/java/org/spout/vanilla/material/item/SpawnEgg.java]: Good idea.

I didn't think the others would work, for some reason. Just tried it, and they do. :P I'll add those.

File: src/main/java/org/spout/vanilla/material/item/SpawnEgg.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+		super(name, id, data);
+		
+		Class<? extends VanillaEntity> controller = org.spout.vanilla.entity.Entity.getByID((int) getData()).getController();
+		
+		Constructor<?> tmp = null;
+		Constructor<?>[] constructors = controller.getConstructors();
+		for (Constructor<?> constructor : constructors) {
+			if (constructor.getParameterTypes().length == 0) {
+				tmp = constructor;
+				break;
```
Comment (Darsstar) [inline src/main/java/org/spout/vanilla/material/item/SpawnEgg.java]: You could return instead of breaking.

Comment (Top-Cat) [inline src/main/java/org/spout/vanilla/material/item/SpawnEgg.java]: But then chosen wouldn't equal tmp ;)

Comment (Darsstar) [inline src/main/java/org/spout/vanilla/material/item/SpawnEgg.java]: Well, I figure he made `tmp` since he made `chosen` final. If he returns he can replace `chosen = tmp` by `chosen = null` and `tmp = constructor` by `chosen = constructor`, right?

Comment (spacechase0) [inline src/main/java/org/spout/vanilla/material/item/SpawnEgg.java]: I originally added `tmp` because when I just had `chosen = constructor`, it was complaining about `chosen` not getting initialized sometimes.

Do I really need to make another commit over such a minor detail though? :P

Comment (Darsstar) [inline src/main/java/org/spout/vanilla/material/item/SpawnEgg.java]: Ask @NinjaZidane, I just wanted to let you know. Oh, sorry for the mail @NinjaZidane :p

**Comment by Zidane** [comment]
Heh...>.>

**Comment by spacechase0** [comment]
Is that a yes or no? :P

**Comment by Zidane** [comment]
@spacechase0 I am sure you can do it really quick...you will be fine haha.

**Comment by spacechase0** [comment]
Fixed, for you picky people. :P
